src/instructions/mod.rs

```diff
@@ -10,9 +10,34 @@ pub mod tables;
 pub mod tlb;
 
 /// Halts the CPU until the next interrupt arrives.
-#[inline(always)]
+#[inline]
 pub fn hlt() {
     unsafe {
         asm!("hlt" :::: "volatile");
     }
 }
+
+/// Emits a 'magic breakpoint' instruction for the [Bochs](http://bochs.sourceforge.net/) CPU
+/// emulator. Make sure to set `magic_break: enabled=1` in your `.bochsrc` file.
+#[inline]
+pub fn bochs_breakpoint() {
+    unsafe {
+        asm!("xchgw %bx, %bx" :::: "volatile");
+    }
+}
+
+/// Gets the current instruction pointer. Note that this is only approximate as it requires a few
+/// instructions to execute.
+#[inline]
+pub fn read_rip() -> u64 {
+    let rip: u64;
+    unsafe {
+        asm!(
+            "call .next
+            .next:
```

Comment:
I'm not sure if inline assembly does support local labels (prefixed with a dot). With `global_asm`, which uses the same syntax, such labels are treated as global labels so that a linker error occurs if another `.next` label is defined elsewhere in the program. Could you try whether this is the case for inline assembly too?

In case it is, it might make sense to manually mangle that label name to something less common. Alternatively, it seems that `lea rax, [rip]` also works to get the `rip` value, without needing any labels.